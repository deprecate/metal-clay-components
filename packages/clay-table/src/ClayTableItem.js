import 'clay-checkbox';
import 'clay-dropdown';
import 'clay-icon';
import 'clay-label';
import 'clay-link';
import 'clay-progress-bar';
import 'clay-sticker';
import {actionsItemsValidator, columnsValidator} from './items_validator';
import {Config} from 'metal-state';
import Component from 'metal-component';
import defineWebComponent from 'metal-web-component';
import dom from 'metal-dom';
import Soy from 'metal-soy';

import templates from './ClayTableItem.soy.js';

/**
 * Metal ClayTableItem component.
 */
class ClayTableItem extends Component {
	/**
	 * Handle input of type `checkbox` for add class `table-active` in tr.
	 * @param {!Event} event
	 * @private
	 */
	handleItemToggled_(event) {
		dom.toggleClasses(this.element, 'table-active');

		this.emit('itemToggled', event);
	}
}

/**
 * State definition.
 * @static
 * @type {!Object}
 */
ClayTableItem.STATE = {
	/**
	 * CSS classes to be applied to the td of the actions menu.
	 * @instance
	 * @memberof ClayTableItem
	 * @type {?string|undefined}
	 * @default undefined
	 */
	actionColumnElementClasses: Config.string(),

	/**
	 * List of items to display in the actions menu.
	 * @instance
	 * @memberof ClayTableItem
	 * @type {?array|undefined}
	 * @default undefined
	 */
	actionItems: actionsItemsValidator,

	/**
	 * List of the columns in the row.
	 * @instance
	 * @memberof ClayTableItem
	 * @type {?array|undefined}
	 * @default undefined
	 */
	columns: columnsValidator,

	/**
	 * Name of the content renderer to use template variants.
	 * @instance
	 * @memberof ClayTableItem
	 * @type {?string|undefined}
	 * @default undefined
	 */
	contentRenderer: Config.string(),

	/**
	 * Flag to indicate if the item is disabled or not.
	 * @instance
	 * @memberof ClayTableItem
	 * @type {?bool}
	 * @default false
	 */
	disabled: Config.bool().value(false),

	/**
	 * Name to be applied to the input element.
	 * @instance
	 * @memberof ClayTableItem
	 * @type {?string|undefined}
	 * @default undefined
	 */
	inputName: Config.string(),

	/**
	 * Value to be applied to the input element.
	 * @instance
	 * @memberof ClayTableItem
	 * @type {?string|undefined}
	 * @default undefined
	 */
	inputValue: Config.string(),

	/**
	 * Flag to indicate if the item is selectable.
	 * @instance
	 * @memberof ClayTableItem
	 * @type {?bool}
	 * @default false
	 */
	selectable: Config.bool().value(false),

	/**
	 * CSS classes to be applied to the td of the selectable.
	 * @instance
	 * @memberof ClayTableItem
	 * @type {?string|undefined}
	 * @default undefined
	 */
	selectableElementClasses: Config.string(),

	/**
	 * Flag to indicate if the item is selected or not.
	 * @instance
	 * @memberof ClayTableItem
	 * @type {?bool}
	 * @default false
	 */
	selected: Config.bool().value(false),

	/**
	 * The path to the SVG spritemap file containing the icons.
	 * @instance
	 * @memberof ClayTableItem
	 * @type {!string}
	 * @default undefined
	 */
	spritemap: Config.string().required(),

	/**
	 * CSS classes to be applied to the td of the sticker.
	 * @instance
	 * @memberof ClayTableItem
	 * @type {?string|undefined}
	 * @default undefined
	 */
	stickerElementClasses: Config.string(),

	/**
	 * Alternate text of the image sticker.
	 * @instance
	 * @memberof ClayTableItem
	 * @type {?string|undefined}
	 * @default undefined
	 */
	stickerImageAlt: Config.string(),

	/**
	 * Source of the image to be rendered inside the sticker.
	 * @instance
	 * @memberof ClayTableItem
	 * @type {?string|undefined}
	 * @default undefined
	 */
	stickerImageSrc: Config.string(),

	/**
	 * Sets the text to be rendered inside sticker.
	 * @instance
	 * @memberof ClayTableItem
	 * @type {?string|undefined}
	 * @default undefined
	 */
	stickerLabel: Config.string(),

	/**
	 * Shape of sticker.  Available shapes are `circle`, `rounded`.
	 * @instance
	 * @memberof ClayTableItem
	 * @type {?string}
	 * @default circle
	 */
	stickerShape: Config.string().value('circle'),

	/**
	 * Sticker style. Available sizes are: `danger`, `dark`, `info`, `light`,
	 * `primary`, `secondary`, `success`, `warning`.
	 * @instance
	 * @memberof ClayTableItem
	 * @type {?string}
	 * @default primary
	 */
	stickerStyle: Config.oneOf([
		'danger',
		'dark',
		'info',
		'light',
		'primary',
		'secondary',
		'success',
		'warning',
	]).value('primary'),
};

defineWebComponent('clay-table-item', ClayTableItem);

Soy.register(ClayTableItem, templates);

export {ClayTableItem};
export default ClayTableItem;
