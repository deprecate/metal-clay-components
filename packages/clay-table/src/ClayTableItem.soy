{namespace ClayTableItem}

/**
 * This renders the component's whole content.
 */
{template .render}
    {@param spritemap: string}
    {@param? actionColumnElementClasses: string}
    {@param? actionItems: list<[
        disabled: bool,
        href: string,
        icon: string,
        label: string,
        quickAction: bool,
        separator: bool
    ]>}
    {@param? columns: list<[
        elementClasses: string,
        href: string,
        label: string,
        labelStyle: string,
        progressBar: [
            maxValue: int,
            minValue: int,
            status: string,
            value: int
        ],
        text: string,
        useEllipse: bool
    ]>}
    {@param? contentRenderer: string}
    {@param? disabled: bool}
    {@param? handleItemToggled_: any}
    {@param? inputName: string}
    {@param? inputValue: string}
    {@param? selectable: bool}
    {@param? selectableElementClasses: string}
    {@param? selected: bool}
    {@param? stickerElementClasses: string}
    {@param? stickerImageAlt: string}
    {@param? stickerImageSrc: string}
    {@param? stickerLabel: string}
    {@param? stickerShape: string}
    {@param? stickerStyle: string}

    {let $attributes kind="attributes"}
        class="
            {if $selectable and $selected}
                table-active
            {/if}
        "
    {/let}

    <tr {$attributes}>
        {call .content}
            {param actionColumnElementClasses: $actionColumnElementClasses /}
            {param actionItems: $actionItems /}
            {param columns: $columns /}
            {param contentRenderer: $contentRenderer /}
            {param disabled: $disabled /}
            {param handleItemToggled_: $handleItemToggled_ /}
            {param inputName: $inputName /}
            {param inputValue: $inputValue /}
            {param selectable: $selectable /}
            {param selectableElementClasses: $selectableElementClasses /}
            {param selected: $selected /}
            {param spritemap: $spritemap /}
            {param stickerElementClasses: $stickerElementClasses /}
            {param stickerImageAlt: $stickerImageAlt /}
            {param stickerImageSrc: $stickerImageSrc /}
            {param stickerLabel: $stickerLabel /}
            {param stickerShape: $stickerShape /}
            {param stickerStyle: $stickerStyle /}
        {/call}
    </tr>
{/template}

/**
 * This renders the content of the row.
 */
{template .content}
    {@param spritemap: string}
    {@param? actionColumnElementClasses: string}
    {@param? actionItems: list<?>}
    {@param? columns: list<?>}
    {@param? contentRenderer: string}
    {@param? disabled: bool}
    {@param? handleItemToggled_: any}
    {@param? inputName: string}
    {@param? inputValue: string}
    {@param? selectable: bool}
    {@param? selectableElementClasses: string}
    {@param? selected: bool}
    {@param? stickerElementClasses: string}
    {@param? stickerImageAlt: string}
    {@param? stickerImageSrc: string}
    {@param? stickerLabel: string}
    {@param? stickerShape: string}
    {@param? stickerStyle: string}

    {if $selectable}
        <td class="{$selectableElementClasses}">
            {call ClayCheckbox.render}
                {param checked: $selected /}
                {param disabled: $disabled /}
                {param events: ['change': $handleItemToggled_] /}
                {param hideLabel: true /}
                {param name: $inputName /}
                {param value: $inputValue /}
            {/call}
        </td>
    {/if}

    {if $stickerLabel or $stickerImageSrc}
        <td class="{$stickerElementClasses}">
            {if $stickerImageSrc}
                {call ClaySticker.render}
                    {param imageAlt: $stickerImageAlt /}
                    {param imageSrc: $stickerImageSrc /}
                    {param shape: $stickerShape ?: 'circle' /}
                {/call}
            {else}
                {call ClaySticker.render}
                    {param label: $stickerLabel /}
                    {param shape: $stickerShape ?: 'circle' /}
                    {param style: $stickerStyle /}
                {/call}
            {/if}
        </td>
    {/if}

    {if $columns}
        {foreach $column in $columns}
            {delcall ClayTableItem.Column variant="$contentRenderer"}
                {param elementClasses: $column.elementClasses /}
                {param href: $column.href /}
                {param label: $column.label /}
                {param labelStyle: $column.labelStyle /}
                {param progressBar: $column.progressBar /}
                {param text: $column.text /}
                {param useEllipse: $column.useEllipse /}
            {/delcall}
        {/foreach}
    {/if}

    {if $actionItems}
        <td class="{$actionColumnElementClasses}">
            {call .quickMenu}
                {param actionItems: $actionItems /}
                {param spritemap: $spritemap /}
            {/call}

            {call ClayActionsDropdown.render}
                {param items: $actionItems /}
                {param spritemap: $spritemap /}
            {/call}
        </td>
    {/if}
{/template}

/**
 * This renders a multiples columns of the row.
 */
{deltemplate ClayTableItem.Column}
    {@param? elementClasses: string}
    {@param? href: string}
    {@param? label: string}
    {@param? labelStyle: string}
    {@param? progressBar: [
        maxValue: int,
        minValue: int,
        status: string,
        value: int
    ]}
	{@param? text: string}
    {@param? useEllipse: bool}

    {let $textContent kind="html"}
        {call .text}
            {param href: $href /}
            {param text: $text /}
        {/call}
    {/let}

	{let $content kind="html"}
		{if $label}
			{call ClayLabel.render}
				{param label: $label /}
				{param style: $labelStyle /}
			{/call}
        {elseif $progressBar}
            {call ClayProgressBar.render}
                {param maxValue: $progressBar.maxValue /}
                {param minValue: $progressBar.minValue /}
                {param status: $progressBar.status /}
                {param value: $progressBar.value /}
            {/call}
		{elseif $text}
            {if $useEllipse}
                <span class="text-truncate" title="{$text}">{$textContent}</span>
            {else}
                {$textContent}
            {/if}
		{/if}
	{/let}

	{let $classes kind="text"}
        {if $elementClasses}
            {$elementClasses}
        {/if}

        {if $useEllipse}
            {sp}table-cell-content
        {/if}
	{/let}

	<td class="{$classes}">
		{$content}
	</td>
{/deltemplate}

/**
 * This renders a text of the column.
 */
{template .text}
    {@param? href: string}
    {@param? text: string}

    {if $text and not $href}
        {$text}
    {elseif $text and $href}
        {call ClayLink.render}
            {param href: $href /}
            {param label: $text /}
        {/call}
    {/if}
{/template}

/**
 * This renders the quick actions menu.
 */
{template .quickMenu}
	{@param actionItems: list<?>}
	{@param spritemap: string}

	<div class="quick-action-menu">
		{foreach $item in $actionItems}
			{if $item.quickAction and $item.icon}
				{call ClayLink.render}
					{param elementClasses: 'quick-action-item' /}
                    {param href: $item.href /}
					{param label kind="html"}
						{call ClayIcon.render}
							{param spritemap: $spritemap /}
							{param symbol: $item.icon /}
						{/call}
					{/param}
				{/call}
			{/if}
		{/foreach}
	</div>
{/template}
