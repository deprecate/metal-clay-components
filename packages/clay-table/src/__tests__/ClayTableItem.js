import ClayTableItem from '../ClayTableItem';

let component;
let spritemap = 'icons.svg';

describe('ClayTableItem', function() {
	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render the default markup', () => {
		component = new ClayTableItem({
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render the ClayTableItem with id', () => {
		component = new ClayTableItem({
			id: 'myId',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render the ClayTableItem with custom element class', () => {
		component = new ClayTableItem({
			elementClasses: 'my-custom-class',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render the ClayTableItem with columns', () => {
		component = new ClayTableItem({
			columns: [
				{
					label: 'Foo',
					useEllipse: true,
				},
				{
					label: 'Bar',
					url: '#1',
				},
				{
					label: 'Foo',
				},
				{
					label: 'Bar',
				},
			],
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render the ClayTableItem with user sticker in the column', () => {
		component = new ClayTableItem({
			columns: [
				{
					label: 'Name',
					useEllipse: true,
				},
			],
			spritemap: spritemap,
			stickerLabel: 'AA',
			stickerStyle: 'primary',
		});

		expect(component).toMatchSnapshot();
	});

	it('should render the ClayTableItem with custom classes in column of sticker', () => {
		component = new ClayTableItem({
			columns: [
				{
					label: 'Name',
					useEllipse: true,
				},
			],
			spritemap: spritemap,
			stickerElementClasses: 'custom-column',
			stickerLabel: 'AA',
			stickerStyle: 'primary',
		});

		expect(component).toMatchSnapshot();
	});

	it('should render ClayTableItem with sticker with image in column', () => {
		component = new ClayTableItem({
			columns: [
				{
					label: 'Name',
					useEllipse: true,
				},
			],
			spritemap: spritemap,
			stickerImageSrc: 'image.png',
			stickerImageAlt: 'Image',
		});

		expect(component).toMatchSnapshot();
	});

	it('should render ClayTableItem with sticker shape `rounded`', () => {
		component = new ClayTableItem({
			columns: [
				{
					label: 'Name',
					useEllipse: true,
				},
			],
			spritemap: spritemap,
			stickerImageSrc: 'image.png',
			stickerImageAlt: 'Image',
			stickerShape: 'rounded',
		});

		expect(component).toMatchSnapshot();
	});

	it('should render the ClayTableItem with progress bar in the column', () => {
		component = new ClayTableItem({
			columns: [
				{
					progressBar: {
						value: 40,
					},
				},
			],
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render the ClayTableItem with label in the column', () => {
		component = new ClayTableItem({
			columns: [
				{
					label: 'Approved',
					labelStyle: 'success',
					type: 'label',
				},
			],
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render the ClayTableItem with action menu', () => {
		component = new ClayTableItem({
			actionItems: [
				{
					label: 'Option 1',
					href: '#1',
				},
				{
					label: 'Option 2',
					separator: true,
					href: '#2',
				},
				{
					label: 'Option 3',
					href: '#3',
				},
			],
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render the ClayTableItem with custom classes in column of `action menu`', () => {
		component = new ClayTableItem({
			actionItems: [
				{
					label: 'Option 1',
					href: '#1',
				},
				{
					label: 'Option 2',
					separator: true,
					href: '#2',
				},
				{
					label: 'Option 3',
					href: '#3',
				},
			],
			actionColumnElementClasses: 'custom-column',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render the ClayTableItem with quick action menu', () => {
		component = new ClayTableItem({
			actionItems: [
				{
					icon: 'trash',
					label: 'Option 1',
					quickAction: true,
					href: '#1',
				},
				{
					icon: 'download',
					label: 'Option 2',
					quickAction: true,
					separator: true,
					href: '#2',
				},
				{
					icon: 'info-circle-open',
					label: 'Option 3',
					quickAction: true,
					href: '#3',
				},
			],
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render the ClayTableItem selectable', () => {
		component = new ClayTableItem({
			columns: [
				{
					label: 'Foo',
					useEllipse: true,
				},
				{
					label: 'Bar',
					href: '#1',
				},
				{
					label: 'Foo',
				},
				{
					label: 'Bar',
				},
			],
			selectable: true,
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render the ClayTableItem with custom classes in columns of selectable', () => {
		component = new ClayTableItem({
			columns: [
				{
					label: 'Foo',
					useEllipse: true,
				},
				{
					label: 'Bar',
					href: '#1',
				},
				{
					label: 'Foo',
				},
				{
					label: 'Bar',
				},
			],
			selectable: true,
			selectableElementClasses: 'custom-column',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render the ClayTableItem selected', () => {
		component = new ClayTableItem({
			columns: [
				{
					label: 'Foo',
					useEllipse: true,
				},
				{
					label: 'Bar',
					href: '#1',
				},
				{
					label: 'Foo',
				},
				{
					label: 'Bar',
				},
			],
			selectable: true,
			selected: true,
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render the ClayTableItem disabled', () => {
		component = new ClayTableItem({
			columns: [
				{
					label: 'Foo',
					useEllipse: true,
				},
				{
					label: 'Bar',
					href: '#1',
				},
				{
					label: 'Foo',
				},
				{
					label: 'Bar',
				},
			],
			selectable: true,
			disabled: true,
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render the ClayTableItem with input name', () => {
		component = new ClayTableItem({
			columns: [
				{
					label: 'Foo',
					useEllipse: true,
				},
				{
					label: 'Bar',
					href: '#1',
				},
				{
					label: 'Foo',
				},
				{
					label: 'Bar',
				},
			],
			inputName: 'name',
			selectable: true,
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render the ClayTableItem with input value', () => {
		component = new ClayTableItem({
			columns: [
				{
					label: 'Foo',
					useEllipse: true,
				},
				{
					label: 'Bar',
					href: '#1',
				},
				{
					label: 'Foo',
				},
				{
					label: 'Bar',
				},
			],
			inputValue: 'value',
			selectable: true,
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a selectable ClayTableItem and emit an event on itemToggle', () => {
		component = new ClayTableItem({
			columns: [
				{
					label: 'Foo',
				},
				{
					label: 'Bar',
					href: '#1',
				},
				{
					label: 'Foo',
				},
				{
					label: 'Bar',
				},
			],
			selectable: true,
			spritemap: spritemap,
		});

		const spy = jest.spyOn(component, 'emit');

		component.element.querySelector('input[type="checkbox"]').click();

		expect(spy).toHaveBeenCalled();
		expect(spy).toHaveBeenCalledWith('itemToggled', expect.any(Object));
	});

	it('should render the ClayTableItem with button', () => {
		component = new ClayTableItem({
			columns: [
				{
					label: 'Foo',
					useEllipse: true,
				},
				{
					label: 'Bar',
					href: '#1',
				},
				{
					label: 'Foo',
				},
				{
					label: 'Bar',
					buttonStyle: 'primary',
					type: 'button',
				},
			],
			inputValue: 'value',
			selectable: true,
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should fail when no spritemap is passed', () => {
		expect(() => {
			component = new ClayTableItem();
		}).toThrow();
	});
});
