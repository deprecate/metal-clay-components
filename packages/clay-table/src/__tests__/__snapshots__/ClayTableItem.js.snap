// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayTableItem should render ClayTableItem with sticker shape \`rounded\` 1`] = `
<tr class="">
  <td>
    <span class="sticker sticker-primary">
      <img alt="Image" class="img-fluid" src="image.png">
    </span>
  </td>
  <td class=" table-cell-content">
    <span class="text-truncate" title="Name">Name</span>
  </td>
</tr>
`;

exports[`ClayTableItem should render ClayTableItem with sticker with image in column 1`] = `
<tr class="">
  <td>
    <span class="sticker sticker-primary rounded-circle">
      <img alt="Image" class="img-fluid" src="image.png">
    </span>
  </td>
  <td class=" table-cell-content">
    <span class="text-truncate" title="Name">Name</span>
  </td>
</tr>
`;

exports[`ClayTableItem should render the ClayTableItem disabled 1`] = `
<tr class="">
  <td>
    <div class="custom-control custom-checkbox">
      <label>
        <input disabled="disabled" class="custom-control-input" ref="input" type="checkbox">
        <span class="custom-control-indicator"></span>
        <span class="custom-control-description sr-only"></span>
      </label>
    </div>
  </td>
  <td class=" table-cell-content">
    <span class="text-truncate" title="Foo">Foo</span>
  </td>
  <td class="">
    <a href="#1">Bar</a>
  </td>
  <td class="">Foo</td>
  <td class="">Bar</td>
</tr>
`;

exports[`ClayTableItem should render the ClayTableItem selectable 1`] = `
<tr class="">
  <td>
    <div class="custom-control custom-checkbox">
      <label>
        <input class="custom-control-input" ref="input" type="checkbox">
        <span class="custom-control-indicator"></span>
        <span class="custom-control-description sr-only"></span>
      </label>
    </div>
  </td>
  <td class=" table-cell-content">
    <span class="text-truncate" title="Foo">Foo</span>
  </td>
  <td class="">
    <a href="#1">Bar</a>
  </td>
  <td class="">Foo</td>
  <td class="">Bar</td>
</tr>
`;

exports[`ClayTableItem should render the ClayTableItem selected 1`] = `
<tr class="table-active">
  <td>
    <div class="custom-control custom-checkbox">
      <label>
        <input checked="" class="custom-control-input" ref="input" type="checkbox">
        <span class="custom-control-indicator"></span>
        <span class="custom-control-description sr-only"></span>
      </label>
    </div>
  </td>
  <td class=" table-cell-content">
    <span class="text-truncate" title="Foo">Foo</span>
  </td>
  <td class="">
    <a href="#1">Bar</a>
  </td>
  <td class="">Foo</td>
  <td class="">Bar</td>
</tr>
`;

exports[`ClayTableItem should render the ClayTableItem with action menu 1`] = `
<tr class="">
  <td>
    <div class="quick-action-menu"></div>
    <div class="dropdown dropdown-action">
      <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-ellipsis-v">
          <title>ellipsis-v</title>
          <use xlink:href="icons.svg#ellipsis-v"></use>
        </svg>
      </button>
      <div class="dropdown-menu">
        <ul class="list-unstyled">
          <li>
            <a class=" dropdown-item" href="#1">Option 1</a>
          </li>
          <li>
            <a class=" dropdown-item" href="#2">Option 2</a>
          </li>
          <li aria-hidden="true" class="dropdown-divider" role="presentation"></li>
          <li>
            <a class=" dropdown-item" href="#3">Option 3</a>
          </li>
        </ul>
      </div>
    </div>
  </td>
</tr>
`;

exports[`ClayTableItem should render the ClayTableItem with columns 1`] = `
<tr class="">
  <td class=" table-cell-content">
    <span class="text-truncate" title="Foo">Foo</span>
  </td>
  <td class="">Bar</td>
  <td class="">Foo</td>
  <td class="">Bar</td>
</tr>
`;

exports[`ClayTableItem should render the ClayTableItem with custom classes in column of \`action menu\` 1`] = `
<tr class="">
  <td class="custom-column">
    <div class="quick-action-menu"></div>
    <div class="dropdown dropdown-action">
      <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-ellipsis-v">
          <title>ellipsis-v</title>
          <use xlink:href="icons.svg#ellipsis-v"></use>
        </svg>
      </button>
      <div class="dropdown-menu">
        <ul class="list-unstyled">
          <li>
            <a class=" dropdown-item" href="#1">Option 1</a>
          </li>
          <li>
            <a class=" dropdown-item" href="#2">Option 2</a>
          </li>
          <li aria-hidden="true" class="dropdown-divider" role="presentation"></li>
          <li>
            <a class=" dropdown-item" href="#3">Option 3</a>
          </li>
        </ul>
      </div>
    </div>
  </td>
</tr>
`;

exports[`ClayTableItem should render the ClayTableItem with custom classes in column of sticker 1`] = `
<tr class="">
  <td class="custom-column">
    <span class="sticker sticker-primary rounded-circle">AA</span>
  </td>
  <td class=" table-cell-content">
    <span class="text-truncate" title="Name">Name</span>
  </td>
</tr>
`;

exports[`ClayTableItem should render the ClayTableItem with custom classes in columns of selectable 1`] = `
<tr class="">
  <td class="custom-column">
    <div class="custom-control custom-checkbox">
      <label>
        <input class="custom-control-input" ref="input" type="checkbox">
        <span class="custom-control-indicator"></span>
        <span class="custom-control-description sr-only"></span>
      </label>
    </div>
  </td>
  <td class=" table-cell-content">
    <span class="text-truncate" title="Foo">Foo</span>
  </td>
  <td class="">
    <a href="#1">Bar</a>
  </td>
  <td class="">Foo</td>
  <td class="">Bar</td>
</tr>
`;

exports[`ClayTableItem should render the ClayTableItem with custom element class 1`] = `<tr class="my-custom-class"></tr>`;

exports[`ClayTableItem should render the ClayTableItem with group 1`] = `
<tr class="table-divider">
  <td colspan="9">Group 1</td>
</tr>
`;

exports[`ClayTableItem should render the ClayTableItem with id 1`] = `<tr class=""></tr>`;

exports[`ClayTableItem should render the ClayTableItem with input name 1`] = `
<tr class="">
  <td>
    <div class="custom-control custom-checkbox">
      <label>
        <input class="custom-control-input" name="name" ref="input" type="checkbox">
        <span class="custom-control-indicator"></span>
        <span class="custom-control-description sr-only"></span>
      </label>
    </div>
  </td>
  <td class=" table-cell-content">
    <span class="text-truncate" title="Foo">Foo</span>
  </td>
  <td class="">
    <a href="#1">Bar</a>
  </td>
  <td class="">Foo</td>
  <td class="">Bar</td>
</tr>
`;

exports[`ClayTableItem should render the ClayTableItem with input value 1`] = `
<tr class="">
  <td>
    <div class="custom-control custom-checkbox">
      <label>
        <input class="custom-control-input" value="value" ref="input" type="checkbox">
        <span class="custom-control-indicator"></span>
        <span class="custom-control-description sr-only"></span>
      </label>
    </div>
  </td>
  <td class=" table-cell-content">
    <span class="text-truncate" title="Foo">Foo</span>
  </td>
  <td class="">
    <a href="#1">Bar</a>
  </td>
  <td class="">Foo</td>
  <td class="">Bar</td>
</tr>
`;

exports[`ClayTableItem should render the ClayTableItem with label in the column 1`] = `
<tr class="">
  <td class="">
    <span class="label label-success">Approved</span>
  </td>
</tr>
`;

exports[`ClayTableItem should render the ClayTableItem with progress bar in the column 1`] = `
<tr class="">
  <td class="">
    <div class="progress-group progress-info">
      <div class="progress">
        <div aria-valuemax="100" aria-valuemin="0" aria-valuenow="40" class="progress-bar" role="progressbar" style="width: 40%;"></div>
      </div>
      <div class="progress-group-addon">40%</div>
    </div>
  </td>
</tr>
`;

exports[`ClayTableItem should render the ClayTableItem with quick action menu 1`] = `
<tr class="">
  <td>
    <div class="quick-action-menu">
      <a class=" quick-action-item" href="#1">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-trash">
          <title>trash</title>
          <use xlink:href="icons.svg#trash"></use>
        </svg>
      </a>
      <a class=" quick-action-item" href="#2">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-download">
          <title>download</title>
          <use xlink:href="icons.svg#download"></use>
        </svg>
      </a>
      <a class=" quick-action-item" href="#3">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-info-circle-open">
          <title>info-circle-open</title>
          <use xlink:href="icons.svg#info-circle-open"></use>
        </svg>
      </a>
    </div>
    <div class="dropdown dropdown-action">
      <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-ellipsis-v">
          <title>ellipsis-v</title>
          <use xlink:href="icons.svg#ellipsis-v"></use>
        </svg>
      </button>
      <div class="dropdown-menu">
        <ul class="list-unstyled">
          <li>
            <a class=" dropdown-item" href="#1">Option 1</a>
          </li>
          <li>
            <a class=" dropdown-item" href="#2">Option 2</a>
          </li>
          <li aria-hidden="true" class="dropdown-divider" role="presentation"></li>
          <li>
            <a class=" dropdown-item" href="#3">Option 3</a>
          </li>
        </ul>
      </div>
    </div>
  </td>
</tr>
`;

exports[`ClayTableItem should render the ClayTableItem with user sticker in the column 1`] = `
<tr class="">
  <td>
    <span class="sticker sticker-primary rounded-circle">AA</span>
  </td>
  <td class=" table-cell-content">
    <span class="text-truncate" title="Name">Name</span>
  </td>
</tr>
`;

exports[`ClayTableItem should render the default markup 1`] = `<tr class=""></tr>`;
